services/director-v2/src/simcore_service_director_v2/models/domains/dynamic_sidecar.py

```diff
@@ -0,0 +1,97 @@
+from pathlib import Path
+from typing import Dict, Any, Optional, List
+
+from pydantic import BaseModel, Field, validator
+
+from ..schemas.constants import UserID
+from models_library.projects import ProjectID
+from models_library.services import SERVICE_KEY_RE
+from models_library.basic_regex import VERSION_RE
+
+
+ComposeSpecModel = Optional[Dict[str, Any]]
+
+
+class PathsMappingModel(BaseModel):
+    inputs_path: Path = Field(
+        ..., description="path where the service expects all the inputs folder"
+    )
+    outputs_path: Path = Field(
+        ..., description="path where the service expects all the outputs folder"
+    )
+    other_paths: List[Path] = Field(
+        [],
+        description="optional list of path which contents need to be saved and restored",
+    )
+
+    @validator("other_paths", always=True)
+    @classmethod
+    def convert_none_to_empty_list(cls, v):
+        return [] if v is None else v
+
+
+class StartDynamicSidecarModel(BaseModel):
+    user_id: UserID
+    project_id: ProjectID
+    service_key: str = Field(
+        ...,
+        description="name of the service",
+        regex=SERVICE_KEY_RE,
+        examples=[
```

Comment:
For examples of models, add better complete examples

```python
class MyModel(BaseModel):

   class Config(BaseConfig):
        schema_extra = {
            example: { ... }
           }
 ```
 and then you can easily test them using ``model_cls_examples`` fixture. Search the latter in the project and you will see a lot of models tests using this.

Reply:
While I agree that it is nice to provide a full example like suggested above. I am not of fan of using the `model_cls_examples` for testing.

I prefer having complete separation between tests and code or documentation in this case.

I will try and provide a full example here.